pygmt/src/triangulate.py

```diff
@@ -0,0 +1,381 @@
+                region=region,
+                verbose=verbose,
+                binary=binary,
+                nodata=nodata,
+                find=find,
+                coltypes=coltypes,
+                header=header,
+                incols=incols,
+                registration=registration,
+                skiprows=skiprows,
+                wrap=wrap,
+                **kwargs,
+            )
+
+    @staticmethod
+    @fmt_docstring
+    @kwargs_to_strings(R="sequence")
+    def delaunay_triples(  # pylint: disable=too-many-arguments,too-many-locals
+        data=None,
+        x=None,
+        y=None,
+        z=None,
+        output_type="pandas",
+        outfile=None,
+        projection=None,
+        verbose=None,
+        binary=None,
+        nodata=None,
+        find=None,
+        coltypes=None,
+        header=None,
+        incols=None,
+        skiprows=None,
+        wrap=None,
+        **kwargs,
+    ):
+        """
+        Delaunay triangle based gridding of Cartesian data.
+
+        Reads in x,y[,z] data and performs Delaunay triangulation, i.e., it
+        finds how the points should be connected to give the most equilateral
+        triangulation possible. If a map projection (give ``region`` and
+        ``projection``) is chosen then it is applied before the triangulation
+        is calculated. The actual algorithm used in the triangulations is
+        either that of Watson [1982] or Shewchuk [1996] [Default if installed;
+        type ``gmt get GMT_TRIANGULATE`` on the command line to see which
+        method is selected). **Note**: For geographic data with global or very
+        large extent you should consider
+        :gmt-docs:`sphtriangulate <sphtriangulate.html>` instead since
+        ``triangulate`` is a Cartesian or small-geographic area operator and is
+        unaware of periodic or polar boundary conditions.
+
+        Must provide either ``data`` or ``x``, ``y``, and ``z``.
+
+        Full option list at :gmt-docs:`triangulate.html`
+
+        Parameters
+        ----------
+        x/y/z : np.ndarray
+            Arrays of x and y coordinates and values z of the data points.
+        data : str or {table-like}
+            Pass in (x, y, z) or (longitude, latitude, elevation) values by
+            providing a file name to an ASCII data table, a 2D
+            {table-classes}.
+        {J}
+        {R}
+        outfile : str or bool or None
+            The name of the output ASCII file to store the results of the
+            histogram equalization in.
+        {V}
+        {b}
+        {d}
+        {e}
+        {f}
+        {h}
+        {i}
+        {s}
+        {w}
+
+        Returns
+        -------
+        ret: pandas.DataFrame or None
+            Return type depends on the ``outfile`` parameter:
+
+            - pandas.DataFrame if ``outfile`` is True or None
+            - None if ``outfile`` is a str (file output is stored in
+              ``outfile``)
```

Comment:
```suggestion
        ret : pandas.DataFrame or numpy.ndarray or None
            Return type depends on ``outfile`` and ``output_type``:

            - None if ``outfile`` is set (output will be stored in file set by
              ``outfile``)
            - :class:`pandas.DataFrame` or :class:`numpy.ndarray` if ``outfile`` is
              not set (depends on ``output_type``)
```
I'll fix this in grdhisteq as well.